src/movies.js

```diff
@@ -1,16 +1,212 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    let allDir = movies.map((elem) => {
+        return elem.director;
+    })
+
+    return allDir;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+function uniqueDirectors(movies) {
+    let dirs = getAllDirectors(movies);
+    let unique = [];
+
+    dirs.forEach((dir) => {
+        if (unique.indexOf(dir) == -1) {
+            unique.push(dir)
+        }
+    })
+
+    return unique;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(arr) {
+    let spielberg = arr.filter((elem) => {
+        return (elem.director == "Steven Spielberg" && elem.genre.indexOf("Drama") != -1);
+    })
+
+    return spielberg.length;
+}
+
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(arr) {
+    if (arr != '') {
+        let ratings = arr.reduce((acc, elem) => {
+            if (elem.rate != undefined) {
+                return acc + elem.rate;
+            }
+            else {
+                return acc;
+            }
+
+        }, 0)
+
+        return Math.round((ratings / arr.length) * 100) / 100;
+    }
+    else {
+        return 0;
+    }
+
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    if (arr != '') {
+        let movies = 0;
+
+        let ratings = arr.reduce((acc, elem) => {
+            if (elem.genre.includes("Drama")) {
+                movies++;
+                return acc + elem.rate
+            }
+            else {
+                return acc;
+            }
+        }, 0)
+
+        if (movies != 0) {
```

Comment:
as best practices remember to use "===" or "!==" when comparing values, unless you are specifically trying to convert data types.